### ISSUE TITLE
All tests that contain inner classes will be skipped if triggered from the test explorer

### ISSUE THREAD

**Issue by bsaby** [Microsoft/vscode-java-test#460]
Result:
![image](https://user-images.githubusercontent.com/22470312/49061813-c321ab00-f24c-11e8-910c-449891fa9ea5.png)

Output
```
[ERROR] @@<TestRunner-{"name":"error", "attributes":{"message":"Failed to parse tests for suite com.example.project.JUnit4Test$A#testAssert1. Details: com.example.project.JUnit4Test$A#testAssert1.", "details":"java.lang.ClassNotFoundException: com.example.project.JUnit4Test$A#testAssert1\r\n\tat java.net.URLClassLoader.findClass(URLClassLoader.java:381)\r\n\tat java.lang.ClassLoader.loadClass(ClassLoader.java:424)\r\n\tat sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)\r\n\tat java.lang.ClassLoader.loadClass(ClassLoader.java:357)\r\n\tat java.lang.Class.forName0(Native Method)\r\n\tat java.lang.Class.forName(Class.java:264)\r\n\tat com.microsoft.java.test.runner.junit4.JUnit4RunnerUtil.getRequestForClasses(JUnit4RunnerUtil.java:80)\r\n\tat com.microsoft.java.test.runner.junit4.JUnit4RunnerUtil.createTestReferences(JUnit4RunnerUtil.java:45)\r\n\tat com.microsoft.java.test.runner.junit4.CustomizedJUnit4CoreRunner.run(CustomizedJUnit4CoreRunner.java:25)\r\n\tat com.microsoft.java.test.runner.junit4.JUnit4Launcher.execute(JUnit4Launcher.java:30)\r\n\tat com.microsoft.java.test.runner.Launcher.main(Launcher.java:53)\r\n"}}-TestRunner>.
[ERROR] 
.
[ERROR] @@<TestRunner-{"name":"error", "attributes":{"message":"Failed to parse tests for suite com.example.project.JUnit4Test#test. Details: com.example.project.JUnit4Test#test.", "details":"java.lang.ClassNotFoundException: com.example.project.JUnit4Test#test\r\n\tat java.net.URLClassLoader.findClass(URLClassLoader.java:381)\r\n\tat java.lang.ClassLoader.loadClass(ClassLoader.java:424)\r\n\tat sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)\r\n\tat java.lang.ClassLoader.loadClass(ClassLoader.java:357)\r\n\tat java.lang.Class.forName0(Native Method)\r\n\tat java.lang.Class.forName(Class.java:264)\r\n\tat com.microsoft.java.test.runner.junit4.JUnit4RunnerUtil.getRequestForClasses(JUnit4RunnerUtil.java:80)\r\n\tat com.microsoft.java.test.runner.junit4.JUnit4RunnerUtil.createTestReferences(JUnit4RunnerUtil.java:45)\r\n\tat com.microsoft.java.test.runner.junit4.CustomizedJUnit4CoreRunner.run(CustomizedJUnit4CoreRunner.java:25)\r\n\tat com.microsoft.java.test.runner.junit4.JUnit4Launcher.execute(JUnit4Launcher.java:30)\r\n\tat com.microsoft.java.test.runner.Launcher.main(Launcher.java:53)\r\n"}}-TestRunner>.
[ERROR] 
.
[ERROR] @@<TestRunner-{"name":"error", "attributes":{"message":"Failed to run Junit tests", "details":"java.lang.NullPointerException\r\n\tat com.microsoft.java.test.runner.common.TestMessageItem.getStacktrace(TestMessageItem.java:44)\r\n\tat com.microsoft.java.test.runner.common.TestMessageItem.<init>(TestMessageItem.java:39)\r\n\tat com.microsoft.java.test.runner.junit4.JUnit4RunnerUtil.getRequestForClasses(JUnit4RunnerUtil.java:91)\r\n\tat com.microsoft.java.test.runner.junit4.JUnit4RunnerUtil.createTestReferences(JUnit4RunnerUtil.java:45)\r\n\tat com.microsoft.java.test.runner.junit4.CustomizedJUnit4CoreRunner.run(CustomizedJUnit4CoreRunner.java:25)\r\n\tat com.microsoft.java.test.runner.junit4.JUnit4Launcher.execute(JUnit4Launcher.java:30)\r\n\tat com.microsoft.java.test.runner.Launcher.main(Launcher.java:53)\r\n"}}-TestRunner>.
[ERROR] 
.

```

**Comment by lnngo** [comment]
I found the same issue but for classes with only one test

**Comment by jdneo** [comment]
The root cause is same as #515. Let's track this in #515  

This should be fixed with #536

**Comment by bsaby** [comment]
Reopen this issue because it can still be reproduced.

**For TestNG project:**
1. Open a TestNG project in Vscode, for example: https://github.com/spring-projects/spring-boot/tree/master/spring-boot-samples/spring-boot-sample-testng
1. Manually add a inner class
1. Trigger test from a level that contains an inner class in test explorer

Result:
All tests will be skipped
![image](https://user-images.githubusercontent.com/22470312/50440549-b23e8500-0931-11e9-85e4-951ae337fe8d.png)

**For Junit5 project**
Trigger test from a level that contains an inner class in test explorer, only the inner class case will be skipped.

**For Junit4 project**
Trigger test from a level that contains an inner class in test explorer, all cases can be run successfully.

There is another test skipped issue of inner class here:
For Junit5 or TestNG project, if triggering test from inner class codelens or inner class level in test explorer, the inner class case will be skipped, but works on Junit4 project.
![image](https://user-images.githubusercontent.com/22470312/50441825-5ecf3580-0937-11e9-8f73-54934e8db8ff.png)
